system/solaris_zone.py

```diff
@@ -0,0 +1,360 @@
+            rc = os.system('ps -z %s -o args|grep "/usr/lib/saf/ttymon.*-d /dev/console" > /dev/null 2>/dev/null' % self.name)
+            if rc == 0:
+                break
+            time.sleep(10)
+            elapsed += 10
+
+    def destroy(self):
+        cmd = '%s -z %s delete -F' % (self.zonecfg_cmd, self.name)
+        (rc, out, err) = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to delete zone. %s' % (out + err))
+
+    def stop(self):
+        cmd = '%s -z %s halt' % (self.zoneadm_cmd, self.name)
+        (rc, out, err) = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to stop zone. %s' % (out + err))
+
+    def exists(self):
+        cmd = '%s -z %s list' % (self.zoneadm_cmd, self.name)
+        (rc, out, err) = self.module.run_command(cmd)
+        if rc == 0:
+            return True
+        else:
+            return False
+
+    def running(self):
+        cmd = '%s -z %s list -p' % (self.zoneadm_cmd, self.name)
+        (rc, out, err) = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to determine zone state. %s' % (out + err))
+
+        if out.split(':')[2] == 'running':
+            return True
+        else:
+            return False
+
+
+    def state_present(self):
+        if self.exists():
+            self.msg.append('zone already exists')
+        else:
+            if not self.module.check_mode:
+                self.create()
+            self.changed = True
+            self.msg.append('zone created')
+
+    def state_running(self):
+        self.state_present()
+        if self.running():
+            self.msg.append('zone already running')
+        else:
+            if not self.module.check_mode:
+                self.boot()
+            self.changed = True
+            self.msg.append('zone booted')
+
+    def state_stopped(self):
+        if self.exists():
+            if self.running():
+                if not self.module.check_mode:
+                    self.stop()
+                self.changed = True
+                self.msg.append('zone stopped')
+            else:
+                self.msg.append('zone not running')
+        else:
+            self.module.fail_json(msg='zone does not exist')
+
+    def state_absent(self):
+        if self.exists():
+            self.state_stopped()
+            if not self.module.check_mode:
+                self.destroy()
+            self.changed = True
+            self.msg.append('zone deleted')
+        else:
+            self.msg.append('zone does not exist')
+
+    def exit_with_msg(self):
+        msg = ', '.join(self.msg)
+        self.module.exit_json(changed=self.changed, msg=msg)
+
+def main():
+    module = AnsibleModule(
+        argument_spec      = dict(
+            name           = dict(required=True),
+            state          = dict(required=True, choices=['running', 'present', 'stopped', 'absent']),
```

Comment:
state is required, so no default is needed, also 'started' would be preferable over running